AssertionError: Padding_idx must be within num_embeddings  MarianModel

### System Info

- `transformers` version: 4.22.2
- Platform: Linux-5.10.133+-x86_64-with-Ubuntu-18.04-bionic
- Python version: 3.7.14
- Huggingface_hub version: 0.10.0
- PyTorch version (GPU?): 1.12.1+cu113 (False)
- Tensorflow version (GPU?): 2.8.2 (False)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: <fill in>
- Using distributed or parallel set-up in script?

### Who can help?

@LysandreJik

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

The code is taken from official huggingface documentation [here](https://huggingface.co/docs/transformers/model_doc/marian#transformers.MarianConfig.example)

from transformers import MarianModel, MarianConfig

#### Initializing a Marian Helsinki-NLP/opus-mt-en-de style configuration
configuration = MarianConfig()

#### Initializing a model from the Helsinki-NLP/opus-mt-en-de style configuration
model = MarianModel(configuration)

#### Accessing the model configuration
configuration = model.config

### Expected behavior

I am trying to initialize MarianModel without using pretrained weights. So instead of this code
```
model_name = "penpen/novel-zh-en"
model = AutoModelForSeq2SeqLM.from_pretrained(model_name)
```` 
I wan to use it with out using pretrained model. Instead I want it to be trained from scratch.  So i used MarianConfig, but it throw the error. 
```
AssertionError: Padding_idx must be within num_embeddings
```

--- a/src/transformers/models/marian/configuration_marian.py
+++ b/src/transformers/models/marian/configuration_marian.py
@@ -110,7 +110,7 @@ class MarianConfig(PretrainedConfig):
 
     def __init__(
         self,
-        vocab_size=50265,
+        vocab_size=58101,
         decoder_vocab_size=None,
         max_position_embeddings=1024,
         encoder_layers=12,

--- a/src/transformers/models/marian/configuration_marian.py
+++ b/src/transformers/models/marian/configuration_marian.py
@@ -43,7 +43,7 @@ class MarianConfig(PretrainedConfig):
 
 
     Args:
-        vocab_size (`int`, *optional*, defaults to 50265):
+        vocab_size (`int`, *optional*, defaults to 58101):
             Vocabulary size of the Marian model. Defines the number of different tokens that can be represented by the
             `inputs_ids` passed when calling [`MarianModel`] or [`TFMarianModel`].
         d_model (`int`, *optional*, defaults to 1024):

